### PR TITLE
fix(search): match escaped unicode content in full-text search - Issue #11538

### LINKED ISSUE #11538: bug: Full-text search fails for non-English text
### Bug Description

Full-text search in the Langfuse UI works correctly for English text but **fails to return results for non-English text** (Arabic, Chinese, Japanese, etc.), even when the traces contain exact matches.

### Expected behavior

Full-text search should return matching traces regardless of the language of the text in the input/output fields.

### Steps to reproduce

1. Create a trace with multilingual input/output:

```python
from langfuse import Langfuse

langfuse_client = Langfuse()

with langfuse_client.start_as_current_observation(
    as_type="span",
    name="multilingual-test",
    input={
        "en": "Hello, how can I help you today?",
        "ar": "مرحبا، كيف يمكنني مساعدتك اليوم؟",
        "zh": "你好，今天我能帮你什么？",
    },
) as span:
    span.update(output={"response": "Test response"})

langfuse_client.flush()
```

2. In the Langfuse UI, go to Traces and use the full-text search
3. Search for `"Hello"` - the trace is returned
4. Search for `"مرحبا"` or `"你好"` - no results returned despite the trace containing this exact text

### Langfuse Cloud or self-hosted?

Self-hosted

### If self-hosted, what version are you running?

v3.141.0 OSS

### SDK and integration versions

_No response_

### Additional information

_No response_

### Are you interested in contributing a fix for this bug?

No

### PR DESCRIPTION
**Pull request by sophyphreak**
## What does this PR do?

  Fixes #11538

  This PR fixes full-text search for non-English trace and observation input/output content when payloads are stored in ClickHouse as JSON strings with `\uXXXX` escapes, which happens through the
  OpenTelemetry / Python SDK ingestion path.

  Previously, searching for raw text like `你好`, `مرحبا`, or `नमस्ते` did not match rows stored as escaped JSON, even though the UI rendered the decoded text correctly. `clickhouseSearchCondition` now also
  searches the JSON-escaped form of non-ASCII queries for `input` and `output` columns, including UTF-16 surrogate pairs for astral characters such as emoji and supplementary-plane CJK.

  ASCII-only searches are unchanged, and plain string columns like `id`, `user_id`, and `name` are not affected. This is query-side only, so existing ClickHouse data becomes searchable without reprocessing.

  No new dependencies.

  Loom Video: N/A, no visual UI change.

  ## Type of change

  - [x] Bug fix (non-breaking change which fixes an issue)

  ## Mandatory Tasks

  - [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

  ## Checklist

  - [x] My code follows the style guidelines of this project
  - [x] I have commented my code where needed, particularly in hard-to-understand areas
  - [x] I have checked if my PR needs changes to the documentation
  - [x] I have added tests that prove my fix is effective
  - [x] I have checked that relevant new and existing tests pass locally

  ## Verification

  Issue-specific coverage passes:

  - `web/src/__tests__/server/multilingual-fulltext-search.servertest.ts`: 45/45 passing
  - `web/src/__e2e__/multilingual-search.spec.ts`: passed as part of the full e2e run
  - `clickhouseSearchCondition.servertest.ts`: 5/5 passing
  - Search-related trace/observation/event repository and table tests passed in the web server test run
  - `packages/shared` TypeScript build: passing
  - `worker` TypeScript build: passing

  Full local suite status after running broader checks:

  - `web` Vitest: 117 passed / 4 failed files, 2444 passed / 14 failed / 7 skipped tests
  - `worker` Vitest: 95 passed / 1 failed files, 1431 passed / 2 failed / 5 skipped tests
  - `web` Playwright e2e: 3 passed / 13 failed

  Known local failures from the broader suite are not in the #11538 search coverage:

  - Billing cycle date math assertions
  - Seed project name mismatch
  - Query builder / metrics assertions against shared ClickHouse test state
  - Worker ingestion integration readback from ClickHouse
  - Auth/create-project e2e flows timing out before reaching app workflows; local dev logs showed dynamic SSO provider decryption errors from local state

<details open><summary><h3>Greptile Summary</h3></summary>

This PR fixes full-text search for non-English content stored in ClickHouse as `\uXXXX`-escaped JSON strings, which is the form produced by Python's `json.dumps` with `ensure_ascii=True` (used by the Langfuse Python SDK / OpenTelemetry ingestion path). The fix adds a query-side escape step so searches for raw Unicode text also match the stored ASCII-escaped representation.

- `clickhouseSearchCondition` now derives a JSON-escaped form of any non-ASCII query and emits an additional `ILIKE {searchStringEscaped: String}` clause for `input`/`output` columns only; ASCII-only queries are completely unaffected.
- `toJsonUnicodeEscaped` encodes BMP characters as `\uXXXX` and astral-plane characters as UTF-16 surrogate pairs, matching Python's serializer output exactly.
- Two new test files are added: a comprehensive server-level integration suite covering 30+ writing systems and edge cases, and an E2E Playwright spec that seeds ClickHouse directly and exercises the full UI search flow.
</details>

<details><summary><h3>Confidence Score: 4/5</h3></summary>

The core fix in search.ts is safe to merge; the two minor issues are confined to the new E2E test helper and do not affect production behaviour.

The production change is a well-scoped, low-risk query-side addition with thorough integration test coverage. The E2E spec has a hardcoded sleep and a loadEnv call that is silently ineffective due to ES module import hoisting, but neither affects correctness of the search fix itself.

web/src/__e2e__/multilingual-search.spec.ts — the loadEnv placement and waitForTimeout should be addressed before the E2E suite is relied upon in CI without the dotenv npm-script prefix.
</details>

<details><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[clickhouseSearchCondition called] --> B{query defined?}
    B -- No --> Z[Return empty query and params]
    B -- Yes --> C[toJsonUnicodeEscaped query]
    C --> D{escapedQuery != query?}
    D -- No ASCII only --> E[hasEscapedVariant = false]
    D -- Yes --> F[hasEscapedVariant = true]
    E --> G[col ILIKE searchString]
    F --> H[col ILIKE searchString OR col ILIKE searchStringEscaped]
    G --> I{searchType?}
    H --> I
    I -- content --> J[input OR output match]
    I -- input --> K[input match only]
    I -- output --> L[output match only]
    I -- id/default --> M[id + user_id + name match]
    J & K & L & M --> O[Return AND conditions + params]
```
</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 2 code review issues. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 2
web/src/__e2e__/multilingual-search.spec.ts:195-197
Hardcoded `waitForTimeout(2000)` is a Playwright anti-pattern that makes the sign-in helper fragile. If the server is slow the 2-second pause may not be enough to detect a sign-in error, and on fast machines it wastes time unconditionally. The error check is already wrapped in `isVisible().catch(() => false)`, so there is no need to sleep before it — the `expect(page).toHaveURL("/")` that follows already waits with its own timeout.

```suggestion
  await submitBtn.click({ timeout: 60_000 }); // "Sign in"
  const err = page.locator(".text-destructive");
```

### Issue 2 of 2
web/src/__e2e__/multilingual-search.spec.ts:117-122
**`loadEnv` call is unreachable-before-imports and therefore ineffective as a fallback.** In ES modules all static `import` declarations are hoisted and fully evaluated before any module-body code runs — so by the time `loadEnv({ path: "../.env" })` executes, `@langfuse/shared/src/server` has already been initialised (and its ClickHouse client constructed) without the env vars loaded from `../.env`. If this spec is invoked directly (without the `dotenv -e ../.env` prefix in the npm script), the shared module will use whatever env vars were set in the shell, not those from the file. Moving `loadEnv` to a `dotenv/config` side-effect import or using a global Playwright setup file would actually achieve the belt-and-suspenders goal described in the comment.

`````

</details>

<sub>Reviews (1): Last reviewed commit: ["Merge branch &#39;main&#39; into bug-#11538-full..."](https://github.com/langfuse/langfuse/commit/fe841e91a238f8c3e134d285495cda79cb4dda37) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=32175408)</sub>

> Greptile also left **1 inline comment** on this PR.

### CODE REVIEW

**Comment by sophyphreak** [comment]
FYI, I am the only committer on this PR. Please let me know if there's something else I need to sign

**Comment by sumerman** [comment]
> FYI, I am the only committer on this PR. Please let me know if there's something else I need to sign

@sophyphreak commits are coming from you and @andrew-palapa 

We won't be able to merge this PR until both of you sign CLA

File: web/src/__e2e__/multilingual-search.spec.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
+/**
```
Comment (sumerman) [inline web/src/__e2e__/multilingual-search.spec.ts]: I think a full integration test is redundant since there's also the servertest.

**Comment by sophyphreak** [comment]
> > FYI, I am the only committer on this PR. Please let me know if there's something else I need to sign
> 
> @sophyphreak commits are coming from you and @andrew-palapa
> 
> We won't be able to merge this PR until both of you sign CLA

I've removed @andrew-palapa as author from any and all commits. (It's just the local name for one of my computers. It doesn't have a separate github account.)

File: web/src/__e2e__/multilingual-search.spec.ts (hunk 1 above)
Comment (sophyphreak) [reply web/src/__e2e__/multilingual-search.spec.ts]: I've removed full integration test file.

**Comment by sophyphreak** [comment]
@sumerman Thanks for your comments and review. Please let me know if you want any more code changes or anything else.

**Comment by sumerman** [comment]
@sophyphreak sorry, it took me some days to return to this. Thank you for your contribution!
